pkg/scaling/executor/scale_jobs.go

```diff
@@ -126,3 +132,89 @@ func (e *scaleExecutor) getRunningJobCount(scaledJob *kedav1alpha1.ScaledJob, ma
 
 	return runningJobs
 }
+
+// Clean up will delete the jobs that is exceed historyLimit
+func (e *scaleExecutor) cleanUp(scaledJob *kedav1alpha1.ScaledJob) error {
+	opts := []client.ListOption{
+		client.InNamespace(scaledJob.GetNamespace()),
+		client.MatchingLabels(map[string]string{"scaledjob": scaledJob.GetName()}),
+	}
+
+	jobs := &batchv1.JobList{}
+	err := e.client.List(context.TODO(), jobs, opts...)
+	if err != nil {
+		e.logger.Info("Can not get job list: ", scaledJob.GetName())
+		return err
+	}
+
+	completedJobs := []batchv1.Job{}
+	failedJobs := []batchv1.Job{}
+	for _, job := range jobs.Items {
+		finishedJobConditionType := e.getFinishedJobConditionType(&job)
+		if finishedJobConditionType == batchv1.JobComplete {
+			completedJobs = append(completedJobs, job)
+			continue
+		}
+		if finishedJobConditionType == batchv1.JobFailed {
+			failedJobs = append(failedJobs, job)
+		}
+	}
+
+	sort.Sort(byCompletedTime(completedJobs))
+	sort.Sort(byCompletedTime(failedJobs))
+
+	var successfulJobsHistoryLimit int32
+	var failedJobsHistoryLimit int32
+	if scaledJob.Spec.SuccessfulJobsHistoryLimit == nil {
+		successfulJobsHistoryLimit = 100 // TODO Default value should be somewhere constant.
+	} else {
+		successfulJobsHistoryLimit = *scaledJob.Spec.SuccessfulJobsHistoryLimit
+	}
+
+	if scaledJob.Spec.FailedJobsHistoryLimit == nil {
+		failedJobsHistoryLimit = 100 // TODO Default value should be somewhere constant.
+	} else {
+		failedJobsHistoryLimit = *scaledJob.Spec.FailedJobsHistoryLimit
+	}
+
+	err = e.deleteJobsWithHistoryLimit(&completedJobs, successfulJobsHistoryLimit)
+	if err != nil {
+		return err
+	}
+	err = e.deleteJobsWithHistoryLimit(&failedJobs, failedJobsHistoryLimit)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (e *scaleExecutor) deleteJobsWithHistoryLimit(jobs *[]batchv1.Job, historyLimit int32) error {
+	if len(*jobs) > int(historyLimit) {
```

Comment:
I think early return is generally more ideomatic in go to avoid nesting. something like 

```go
if len(jobs) <= historyLimit {
	return nil
}

// rest
```

Reply:
Agree, it is much `go style` I love it. Done